File: monsoon3.0/src/main/java/wtf/monsoon/impl/event/EventPreMotion.java
package wtf.monsoon.impl.event;

import lombok.Data;
import lombok.NonNull;
import wtf.monsoon.api.event.Event;

@Data
public class EventPreMotion extends Event {

    @NonNull
    private double x, y, z;
    @NonNull
    private float yaw, pitch;
    @NonNull
    private boolean onGround;

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/event/EventPreMotion.kt
package wtf.monsoon.client.event

import me.bush.eventbuskotlin.Event
import net.minecraft.client.gui.ScaledResolution

data class EventPreMotion(var x: Double, var y: Double, var z: Double, var yaw: Float, var pitch: Float, var onGround: Boolean) : Event() {
    override val cancellable: Boolean = true
}